rocksdb_replicator/replicated_db.cpp

```diff
@@ -139,11 +150,35 @@ rocksdb::Status RocksDBReplicator::ReplicatedDB::Write(
     // TODO(bol): This potentially could block all worker threads. We may
     // consider having a dedicated set of worker threads for admin requests,
     // and/or provide async write API when this turns out to be a problem.
-    if (!max_seq_no_acked_.wait(cur_seq_no, FLAGS_replicator_timeout_ms)) {
+    if (!max_seq_no_acked_.wait(cur_seq_no, current_replicator_timeout_ms_.load())) {
       incCounter(kReplicatorWriteWaitTimedOut, 1, db_name_);
       LOG(ERROR) << "Failed to receive ack from follower, timing out for " << db_name_;
+      numConsecutiveReplTimeout_ ++;
+
+      // enter degradation mode if there has been consecutive timeouts waiting for the follower ACK.
+      // This allows us to use a smaller wait timeout to fail fast, instead of blocking for long.
+      if (numConsecutiveReplTimeout_.load() == FLAGS_replicator_consecutive_ack_timeout_before_degradation
```

Comment:
Should we make this check to be greater than? Since this is multithreaded, there's a possibility it goes from 99 -> 101 before the check completes

Reply:
Good point, updated